src/test/fuzz/bitset.cpp

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <span.h>
+#include <test/fuzz/util.h>
+#include <test/util/xoroshiro128plusplus.h>
+#include <util/bitset.h>
+
+#include <bitset>
+#include <vector>
+
+namespace {
+
+/** Pop the first byte from a Span<const uint8_t>, and return it. */
+uint8_t ReadByte(Span<const uint8_t>& buffer)
+{
+    if (buffer.empty()) return 0;
+    uint8_t ret = buffer.front();
+    buffer = buffer.subspan(1);
+    return ret;
+}
+
+/** Perform a simulation fuzz test on BitSet type S. */
+template<typename S>
+void TestType(Span<const uint8_t> buffer)
+{
+    XoRoShiRo128PlusPlus rng(buffer.size() + 0x10000 * S::Size());
+
+    using Sim = std::bitset<S::Size()>;
+    // Up to 4 real BitSets (initially 2).
+    std::vector<S> real(2);
+    // Up to 4 std::bitsets with the same corresponding contents.
+    std::vector<Sim> sim(2);
+
+    /* Compare sim[idx] with real[idx], using all inspector operations. */
+    auto compare_fn = [&](unsigned idx) {
+        /* iterators and operator[] */
+        auto it = real[idx].begin();
+        unsigned first = S::Size();
+        unsigned last = S::Size();
+        for (unsigned i = 0; i < S::Size(); ++i) {
+            bool match = (it != real[idx].end()) && *it == i;
+            assert(sim[idx][i] == real[idx][i]);
+            assert(match == real[idx][i]);
+            assert((it == real[idx].end()) != (it != real[idx].end()));
+            if (match) {
+                ++it;
+                if (first == S::Size()) first = i;
+                last = i;
+            }
+        }
+        assert(it == real[idx].end());
+        assert(!(it != real[idx].end()));
+        /* Any / None */
+        assert(sim[idx].any() == real[idx].Any());
+        assert(sim[idx].none() == real[idx].None());
+        /* First */
+        if (sim[idx].any()) {
+            assert(first == real[idx].First());
+            assert(last == real[idx].Last());
+        }
+        /* Count */
+        assert(sim[idx].count() == real[idx].Count());
+    };
+
+    LIMITED_WHILE(buffer.size() > 0, 1000) {
+        int command = ReadByte(buffer) % 64;
+        unsigned args = ReadByte(buffer);
+        unsigned dest = ((args & 7) * sim.size()) >> 3;
+        unsigned src = (((args >> 3) & 7) * sim.size()) >> 3;
+        unsigned aux = (((args >> 6) & 3) * sim.size()) >> 2;
+        // Loop until command reaches 0. This way, every iteration will cause one operation
```

Comment:
I like the comment in the VecDeque PR better if you'd care to just copy it here: https://github.com/bitcoin/bitcoin/pull/30161/commits/ecb278bb19c53b007380e262fa86d809255eeb49#diff-1f28310f089eac74210ecbc0db2083279282ade962dc464bd31448480dc5563dR59